src/FileCache.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Aws;
+
+class FileCache implements CacheInterface
+{
+    private $cacheDir;
+
+    public function __construct()
+    {
+        $this->cacheDir = (new JsonCompiler)->getCacheDir();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function set($key, $data, $ttl = 0)
+    {
+        $toSave = ['data' => $data];
+        $path = $this->getCachePath($key);
+        if ($ttl) {
+            $toSave['expiration'] = time() + $ttl;
+        }
+
+        if (@mkdir(dirname($path), 0777, true)) {
+            file_put_contents($path, \serialize($toSave), LOCK_EX);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($key)
+    {
+        unlink($this->getCachePath($key));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key)
+    {
+        if ($retrieved = @file_get_contents($this->getCachePath($key))) {
```

Comment:
I think it might be better to first do a `file_exists` and then a regular `file_get_contents` call.